plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -316,21 +311,28 @@ public static void wrapListenerToHandleSearchIndexNotFound(Exception e, ActionLi
         }
     }
 
-    public static boolean isMultiTenant(ClusterService clusterService, Settings settings) {
-        isMultiTenant = ML_COMMONS_MULTI_TENANCY_ENABLED.get(settings);
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(ML_COMMONS_MULTI_TENANCY_ENABLED, it -> isMultiTenant = it);
-        return isMultiTenant;
-    }
-
-    public static String getTenantID(ClusterService clusterService, Settings settings, RestRequest restRequest) {
-        if (isMultiTenant(clusterService, settings)) {
+    public static String getTenantID(Boolean isMultiTenancyEnabled, RestRequest restRequest) {
+        if (isMultiTenancyEnabled) {
             Map<String, List<String>> headers = restRequest.getHeaders();
             if (headers != null) {
-                String tenantId = restRequest.getHeaders().get(Constants.TENANT_ID).get(0);
-                if (tenantId == null) {
-                    throw new OpenSearchStatusException("Tenant ID can't be null", RestStatus.INTERNAL_SERVER_ERROR);
+                if (headers.containsKey(Constants.TENANT_ID)) {
+                    List<String> tenantIdList = headers.get(Constants.TENANT_ID);
+                    if (tenantIdList != null && !tenantIdList.isEmpty()) {
+                        String tenantId = tenantIdList.get(0);
+                        if (tenantId != null) {
+                            return tenantId;
+                        } else {
+                            throw new OpenSearchStatusException("Tenant ID can't be null", RestStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Should this be `INTERNAL_SERVER_ERROR`  or `UNAUTHORIZED`

Reply:
Check other places as well

Reply:
`FORBIDDEN`